lib/action_controller/serialization.rb

```diff
@@ -30,7 +30,7 @@ def get_serializer(resource, options = {})
         serializable_resource.serialization_scope ||= serialization_scope
         serializable_resource.serialization_scope_name = _serialization_scope
         begin
-          serializable_resource.adapter
+          serializable_resource.tap(&:adapter)
```

Comment:
Spec that breaks is without calling the adapter is `ActionController::Serialization::ImplicitSerializerTest#test_render_json_array_object_without_serializer` with `ActiveModel::Serializer::CollectionSerializer::NoSerializerError:   No serializer found for resource: {:error=>"Result is Invalid"}``